core/src/layers/prometheus.rs

```diff
@@ -93,47 +89,123 @@ use crate::*;
 ///     println!("## Prometheus Metrics");
 ///     println!("{}", String::from_utf8(buffer.clone()).unwrap());
 ///     Ok(())
-/// }
+/// # }
 /// ```
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct PrometheusLayer {
-    registry: Registry,
-    operation_duration_seconds_buckets: Vec<f64>,
-    operation_bytes_buckets: Vec<f64>,
-    path_label_level: usize,
+    interceptor: PrometheusInterceptor,
```

Comment:
Hi, I don't think this change is necessary. We can implement `Default` for `PrometheusLayer` to use `default_registry()` by default, and offer a `new(registry: Registry)` method for users who want to use their own registry.

Reply:
Actually, I don't think we should hold the registry in `PrometheusLayer` struct (although the `Registry` type uses Arc internally)